src/Servers/Kestrel/Transport.Quic/src/WebHostBuilderQuicExtensions.cs

```diff
@@ -16,19 +16,25 @@ public static class WebHostBuilderQuicExtensions
     {
         public static IWebHostBuilder UseQuic(this IWebHostBuilder hostBuilder)
         {
-            if (!QuicImplementationProviders.Default.IsSupported)
+            if (QuicImplementationProviders.Default.IsSupported)
             {
-                throw new NotSupportedException("QUIC is not supported or enabled on this platform. See https://aka.ms/aspnet/kestrel/http3reqs for details.");
+                return hostBuilder.ConfigureServices(services =>
+                {
+                    services.AddSingleton<IMultiplexedConnectionListenerFactory, QuicTransportFactory>();
+                });
             }
-            return hostBuilder.ConfigureServices(services =>
-            {
-                services.AddSingleton<IMultiplexedConnectionListenerFactory, QuicTransportFactory>();
-            });
+
+            return hostBuilder;
         }
 
         public static IWebHostBuilder UseQuic(this IWebHostBuilder hostBuilder, Action<QuicTransportOptions> configureOptions)
         {
-            return hostBuilder.UseQuic().ConfigureServices(services =>
+            return hostBuilder.UseQuic().ConfigureQuic(configureOptions);
+        }
+
+        public static IWebHostBuilder ConfigureQuic(this IWebHostBuilder hostBuilder, Action<QuicTransportOptions> configureOptions)
```

Comment:
Why a different name? Does this match what is in sockets transport?

Reply:
This mimics the pattern we use for Kestrel:

Reply:
Removed